sdk/nodejs/runtime/invoke.ts

```diff
@@ -67,7 +68,14 @@ const providerproto = require("../proto/provider_pb.js");
  */
 export function invoke(tok: string, props: Inputs, opts: InvokeOptions = {}): Promise<any> {
     if (opts.async) {
-        // Use specifically requested async invoking.  Respect that.
+        // User specifically requested async invoking.  Respect that.
+        return invokeAsync(tok, props, opts);
+    }
+
+    const config = new Config("pulumi");
+    const noSyncCalls = config.getBoolean("noSyncCalls");
```

Comment:
On naming - perhaps:
* `disableSyncInvokes`
* `noSyncInvokes`
* `onlyAsyncInvokes`

The problem is none of these really match how the user thinks about it - that their `.getFoo` will now return only a promise, not an actual value.

Thoughts?  @pgavlin @CyrusNajmabadi

Reply:
> @pgavlin Any other option you think we should consider here?

Yes. We could pretty easily update `StackReference`'s `getOutputSync` to use an invoke. It's just some work in the builtin provider.

Reply:
I used "calls" because we describe this in our docs like so: `Synchronous call made to “X” with an unregistered provider`

So "noSyncCalls" seemed to match up the most to that terminology.  I would also be ok with onlyAsyncCalls.  I would avoid "invoke" in the wording as that doens't seem to be how we describe things elsewhere.

Reply:
The concern I have is that `noSyncCalls` doesn't make it clear this has anything to do with `.getXYZ` methods.  The error message you quote above makes more sense because X will in practice be `getAmi` or similar which will make it clear.  I don't have a great suggestion, but I do think `noSyncCalls` is very vague.